dcos/dcos-ui
# DCOS_OSS-4055: Implement reset functionality for delay

Implement reset functionality for services
that are delayed.

Closes https://jira.mesosphere.com/browse/DCOS_OSS-4055

## Testing
1. Create a service that should be running.
Example JSON:
```
{
  "id": "/non-delayed",
  "instances": 1,
  "portDefinitions": [],
  "container": {
    "type": "MESOS",
    "volumes": []
  },
  "cpus": 0.1,
  "mem": 128,
  "requirePorts": false,
  "networks": [],
  "healthChecks": [],
  "fetch": [],
  "constraints": [],
  "cmd": "sleep 1000"
}
```
2. Verify that its actions dropdown both from the services table and from service detail view doesn't contain "Reset Delay".
3. Create a service that will be delayed.
Example JSON from the jira:
```
{
"id": "/delayed",
"backoffFactor": 10,
"backoffSeconds": 1,
"cmd": "exit 1",
"container": {
"type": "MESOS",
"volumes": []
},
"cpus": 0.1,
"disk": 0,
"instances": 10,
"maxLaunchDelaySeconds": 3600,
"mem": 10,
"gpus": 0,
"networks": [
{
"mode": "host"
}
],
"portDefinitions": [],
"requirePorts": false,
"upgradeStrategy": {
"maximumOverCapacity": 1,
"minimumHealthCapacity": 1
},
"killSelection": "YOUNGEST_FIRST",
"unreachableStrategy": {
"inactiveAfterSeconds": 0,
"expungeAfterSeconds": 0
},
"healthChecks": [],
"fetch": [],
"constraints": []
}
```
4. Verify that its actions dropdown both from the services table and from service detail view contains "Reset Delay". Verify that in the detail view, the debug tab also has a "Reset Delay" button. Verify that they all work, showing success or error toast.
5. Verify the same for pods.
Example JSON:
non-delayed
```
{
  "id": "/non-delayed-pod",
  "containers": [
    {
      "name": "container-1",
      "resources": {
        "cpus": 0.1,
        "mem": 128
      },
      "exec": {
        "command": {
          "shell": "sleep 1000"
        }
      }
    }
  ],
  "scaling": {
    "instances": 1,
    "kind": "fixed"
  },
  "networks": [
    {
      "mode": "host"
    }
  ],
  "volumes": [],
  "fetch": [],
  "scheduling": {
    "placement": {
      "constraints": []
    }
  }
}
```
delayed
```
{
  "id": "/delayed-pod",
  "version": "2019-05-28T11:20:07.349Z",
  "containers": [
    {
      "name": "container-1",
      "resources": {
        "cpus": 0.1,
        "mem": 10,
        "disk": 0,
        "gpus": 0
      },
      "exec": {
        "command": {
          "shell": "exit 1"
        }
      }
    }
  ],
  "networks": [
    {
      "mode": "host"
    }
  ],
  "scaling": {
    "instances": 10,
    "kind": "fixed"
  },
  "scheduling": {
    "backoff": {
      "backoff": 1,
      "backoffFactor": 1.15,
      "maxLaunchDelay": 300
    },
    "upgrade": {
      "minimumHealthCapacity": 1,
      "maximumOverCapacity": 1
    },
    "killSelection": "YOUNGEST_FIRST",
    "unreachableStrategy": {
      "inactiveAfterSeconds": 0,
      "expungeAfterSeconds": 0
    },
    "placement": {
      "constraints": []
    }
  },
  "executorResources": {
    "cpus": 0.1,
    "mem": 32,
    "disk": 10
  },
  "volumes": [],
  "fetch": []
}
```
6. Verify that the added tests make sense.

## Trade-offs
https://github.com/dcos/dcos-ui/pull/3917/files?file-filters%5B%5D=.js&file-filters%5B%5D=.ts&owned-by%5B%5D=#r288977721

## Dependencies
https://github.com/dcos/dcos-ui/pull/3914

## Screenshots
![Peek 2019-05-30 15-57](https://user-images.githubusercontent.com/40791275/58634414-e9a3ca00-82f3-11e9-9f02-2f532ce7c57c.gif)
![Peek 2019-05-30 16-01](https://user-images.githubusercontent.com/40791275/58634512-27a0ee00-82f4-11e9-8b0f-0ac07dfccc53.gif)
![Peek 2019-05-30 16-07](https://user-images.githubusercontent.com/40791275/58634874-0ee50800-82f5-11e9-8609-41249a688971.gif)
